measurements/api/measurements.py

```diff
@@ -143,26 +151,40 @@ def get_measurement(measurement_id):
             Autoclaved.filename.label('a_filename'),
             Autoclaved.autoclaved_no.label('a_autoclaved_no'),
 
-    ).filter(Measurement.id == measurement_id)\
+    ).filter(Measurement.msm_no == msm_no)\
         .join(Report, Report.report_no == Measurement.report_no)\
         .join(Autoclaved, Autoclaved.autoclaved_no == Report.autoclaved_no)
     try:
         msmt = q.one()
     except exc.MultipleResultsFound:
         current_app.logger.warning("Duplicate rows for measurement_id: %s" % measurement_id)
         msmt = q.first()
+    except exc.NoResultFound:
+        # XXX we should actually return a 404 here
+        raise BadRequest("No measurement found")
 
     r = requests.get(urljoin(current_app.config['AUTOCLAVED_BASE_URL'], msmt.a_filename),
             headers={"Range": "bytes={}-{}".format(msmt.frame_off, msmt.frame_off+msmt.frame_size)}, stream=True)
 
     # XXX use for streaming support lz4framed.Decompressor
     # @darkk how big can these lz4 frames even become? Should this be a concern?
     msmt_data = lz4framed.decompress(r.raw.read())[msmt.intra_off:msmt.intra_off+msmt.intra_size]
+
+    # XXX do we also want to replace on the fly the measurement_id inside of
+    # the report itself?
```

Comment:
I don't.
I think that client-generated `measurement_id` is valuable for idempotency. It's just not 100% unique. It's like `Message-Id` in emails :)

Reply:
If we don't there is a mismatch between this and what we return below. I think that having `measurement_id` being user generated has basically null value to an end user of our data, since they can't really do anything with it. I would at this point either strip it from the data, or rewrite it with something slightly more meaningful.